src/game.cpp

```diff
@@ -1837,6 +1841,7 @@ static void getPlayerNames()
 // UserSaveGame ... this is true when you are loading a players save game
 bool loadGame(const char *pGameToLoad, bool keepObjects, bool freeMem, bool UserSaveGame)
 {
+	ScriptMapData data;  // Only valid if generating from 'game.js'.
```

Comment:
Since `data` is used outside the scope where it's initialised, it couldn't be initialised during the declaration. I'm unsure whether it was best to declare it at the top of the function or just above the `if` where it's initialised.

Reply:
In this case I think it should be initialized just above the `if`, as currently there are the 400LOC between the declaration and the initialization. Though, this is just nit-picking given how horrifying the whole method is. There are much more likely sources for bugs than one prematurely initialized variable 😄

Reply:
Oh, another reason to put it at the top that I just remembered when trying to move it further down now was lots of `error: jump to label ‘error’ crosses initialization of ‘ScriptMapData data’` errors. So the whole method is too horrifying to even compile, unless I put it at (or very near) the top.